src/Hack.sol

```diff
@@ -626,55 +629,74 @@ contract WinnerTakesAll {
         _;
     }
 
-    /// @dev Creates new rounds.
-    /// @param _numberOfRounds The number of rounds to create.
-    function createNewRounds(uint256 _numberOfRounds) external {
-        for (uint256 i = 0; i < _numberOfRounds; i++) {
-            rounds.push();
+    /// @dev Creates new lotteries.
+    /// @param numberOfLotteries The number of lotteries to create.
+    function createNewLotteries(uint256 numberOfLotteries) external {
+        for (uint256 i = 0; i < numberOfLotteries; i++) {
+            lotteries.push();
         }
     }
 
+    /// @dev Buys a ticket for a participant.
+    /// @param lotteryIndex The index of the round concerned.
+    function buyTicketForLottery(uint256 lotteryIndex) external payable {
+        require(msg.value == 1 ether, "wrong value");
+        lotteries[lotteryIndex].ticketNumber++;
+        lotteries[lotteryIndex].ticketDistribution[msg.sender].push(lotteries[lotteryIndex].ticketNumber);
```

Comment:
would be better to store the update value of `lotteries[lotteryIndex].ticketNumber` in temporary variable no?

Reply:
yes I agree.